ArduCopter/mode_throw.cpp

```diff
@@ -266,23 +266,42 @@ bool ModeThrow::throw_detected()
         changing_height = inertial_nav.get_velocity_z_up_cms() > THROW_VERTICAL_SPEED;
     }
 
-    // Check the vertical acceleraton is greater than 0.25g
+    // Check the vertical acceleration is greater than 0.25g
     bool free_falling = ahrs.get_accel_ef().z > -0.25 * GRAVITY_MSS;
+    
+    // Check if the drone is spinning at a speed greater than 2 rounds per second
+    bool frisbee_throw = fabsf(ahrs.get_gyro().z) > 2.0  * M_2PI;
 
     // Check if the accel length is < 1.0g indicating that any throw action is complete and the copter has been released
     bool no_throw_action = copter.ins.get_accel().length() < 1.0f * GRAVITY_MSS;
 
-    // High velocity or free-fall combined with increasing height indicate a possible air-drop or throw release
-    bool possible_throw_detected = (free_falling || high_speed) && changing_height && no_throw_action;
 
-    // Record time and vertical velocity when we detect the possible throw
+    bool possible_throw_detected = false;
+    switch ((int) g2.throw_type) {
+    case (int) ThrowType::Frisbee :
```

Comment:
Better to cast the param to `ThrowType` removing the need to cast each case. 
```suggestion
    switch ((ThrowType) g2.throw_type) {
    case ThrowType::Frisbee :
```

Reply:
Thanks for the suggestion, i'll make the change.